packages/commerce-sdk-react/src/hooks/ShopperBaskets/mutation.test.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {act} from '@testing-library/react'
+import nock from 'nock'
+import {DEFAULT_TEST_HOST, renderHookWithProviders} from '../../test-utils'
+import {getQueryKeysMatrix, ShopperBasketMutationType, useShopperBasketsMutation} from './mutation'
+import {useBasket} from './query'
+import {useCustomerBaskets} from '../ShopperCustomers/query'
+
+const CUSTOMER_ID = 'CUSTOMER_ID'
+const BASKET_ID = 'BASKET_ID'
+
+jest.mock('../../auth/index.ts', () => {
+    return jest.fn().mockImplementation(() => ({
+        ready: jest.fn().mockResolvedValue({access_token: '123'})
+    }))
+})
+
+jest.mock('../useCustomerId.ts', () => {
+    return jest.fn().mockReturnValue(CUSTOMER_ID)
+})
+
+// TODO: should this be in test-utils? So it can be reused across test suites
+const mockMutationEndpoints = (matchingPath: string, options?: {errorResponse: number}) => {
+    const responseStatus = options?.errorResponse ? options.errorResponse : 200
+
+    nock(DEFAULT_TEST_HOST)
+        .patch((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(responseStatus, {test: 'new data'})
+        .put((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(responseStatus, {test: 'new data'})
+        .post((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(responseStatus, {test: 'new data'})
+        .delete((uri) => {
+            return uri.includes(matchingPath)
+        })
+        .reply(options?.errorResponse ? options.errorResponse : 204)
+}
+
+type MutationPayloads = {
+    [key in ShopperBasketMutationType]?: {body: any; parameters: any}
+}
+const mutationPayloads: MutationPayloads = {
+    updateBasket: {parameters: {basketId: BASKET_ID}, body: {}},
+    updateBillingAddressForBasket: {parameters: {basketId: BASKET_ID}, body: {}},
+    deleteBasket: {parameters: {basketId: BASKET_ID}, body: {}}
```

Comment:
It looks like the underlying isomorphic sdk checks the `parameters` only (and not `body`). So we can have `body` as simply empty object.

Reply:
@yunakim714 ^